### PR TITLE
Properly look for git on Windows

### PR DESCRIPTION
**Pull request by gmittert**
- Look for git.exe instead of git
- The Path environment variable on Windows is ';' delimited

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please test

File: Sources/TSCBasic/misc.swift [hunk 1]
```diff
@@ -48,9 +48,14 @@ public func getEnvSearchPaths(
...
-    return (pathString ?? "").split(separator: ":").map(String.init).compactMap({ pathString in
+  #if os(Windows)
+    let sep: Character = ";"
+  #else
+    let sep: Character = ":"
+  #endif
+    return (pathString ?? "").split(separator: sep).map(String.init).compactMap({ pathString in
         // If this is an absolute path, we're done.
-        if pathString.first == "/" {
+        if pathString.isAbsolutePath {
```
Comment (aciidb0mb3r) [inline Sources/TSCBasic/misc.swift]: This is only available on `NSString`. I've been thinking about adding some new APIs for such common path operations. Maybe we should add a `Path` struct and expose a static method that can answer this question. It'll also be useful for some other common operations like getting the `extension`. We can use AbsolutePath/RelativePath under the hood.

Comment (gmittert) [reply Sources/TSCBasic/misc.swift]: To make sure I understand you, essentially Foundation's [NSPathUtilities](https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSPathUtilities.swift) except they don't go through NSString on Darwin?

Comment (aciidb0mb3r) [reply Sources/TSCBasic/misc.swift]: The problem is that they only go through NSString on Darwin. These properties are unavailable on String. I don't know why corelibs-foundation allows these properties on String. cc @millenomi

Comment (gmittert) [reply Sources/TSCBasic/misc.swift]: My impression is that NSString is bridged to String on all platforms. However, on Darwin, it goes through the Objective C NSString while on Linux and Windows it goes through the swift-corelibs-foundation NSString. Is that not correct?

**Comment by tomerd** [comment]
@compnerd is this still relevant given your recent Windows work?

**Comment by gmittert** [comment]
Looks like a similar fix has already been added. Closing.

**Pull request closed without merging**